src/main/java/io/vertx/core/http/impl/HttpUtils.java

```diff
@@ -52,6 +52,172 @@
   private HttpUtils() {
   }
 
+  private static int indexOfSlash(CharSequence str, int start) {
+    for (int i = start; i < str.length(); i++) {
+      if (str.charAt(i) == '/') {
+        return i;
+      }
+    }
+
+    return -1;
+  }
+
+  private static boolean matches(CharSequence path, int start, String what) {
+    return matches(path, start, what, false);
+  }
+
+  private static boolean matches(CharSequence path, int start, String what, boolean exact) {
+    if (exact) {
+      if (path.length() - start != what.length()) {
+        return false;
+      }
+    }
+
+    if (path.length() - start >= what.length()) {
+      for (int i = 0; i < what.length(); i++) {
+        if (path.charAt(start + i) != what.charAt(i)) {
+          return false;
+        }
+      }
+      return true;
+    }
+
+    return false;
+  }
+
+  /**
+   * Removed dots as per <a href="http://tools.ietf.org/html/rfc3986#section-5.2.4>rfc3986</a>.
+   *
+   * There are 2 extra transformations that are not part of the spec but kept for backwards compatibility:
+   *
+   * double slash // will be converted to single slash and the path will always start with slash.
+   *
+   * @param path raw path
+   * @return normalized path
+   */
+  public static String removeDots(CharSequence path) {
```

Comment:
I reviewed the same kind of code for @pmlopes in vertx-web recently. Can you check with him if we can keep just one?